yesod-core/src/Yesod/Core/Class/Breadcrumbs.hs

```diff
@@ -28,4 +29,8 @@ breadcrumbs = do
     go back Nothing = return back
     go back (Just this) = do
         (title, next) <- breadcrumb this
-        go ((this, title) : back) next
+        if this `elem` map fst back
```

Comment:
I think it would make more sense to perform this check _before_ calling `breadcrumb`. There's a minor performance improvement (not something I care about here), but IMO the code is easier to read. Actually, I'd probably go for a guard here:

```haskell
go back (Just this)
  | this `elem` map fst back = error ...
  | otherwise = do ...
```

Reply:
Oh, doh! This makes a lot of sense.
I tried this out in the latest commit. It does require another constraint but I'm pretty sure that a `Show` instance is also generated.

Reply:
I'll merge with that constraint, and if anyone complains we can remove it later. Thanks!